packages/synapse-react-client/src/components/widgets/facet-nav/PlotsContainer.tsx

```diff
@@ -0,0 +1,379 @@
+    )?.plotType
+    return plotType ?? 'PIE'
+  }
+
+  // Wrap in useCallback
+  // switch to
+  const isPlotInState = useCallback(
+    (plotId: PlotIdentifier): boolean => {
+      // plotUiStateArray.find...
+      //return plotUiStateArray && plotUiStateArray.find(plotId)
+      return !!plotUiStateArray.find(plot => {
+        if ('__custom' in plot.plotId && '__custom' in plotId) {
+          return plot.plotId.title === plotId.title
+        } else if (!('__custom' in plotId) && !('__custom' in plot.plotId)) {
+          return facetObjectMatchesDefinition(plot.plotId, plotId)
+        } else {
+          return false
+        }
+      })
+    },
+    [plotUiStateArray],
+  )
+
+  // Wrap in useCallback
+  const getCombinedNewPlots = useCallback(
+    (
+      customPlots: QueryWrapperSynapsePlotProps[],
+      facetNavPanelPropsArray: Pick<
+        FacetNavPanelProps,
+        'applyChangesToFacetFilter' | 'applyChangesToGraphSlice' | 'facetToPlot'
+      >[],
+    ): UiPlotState[] => [
+      ...(customPlots ?? []).map((plotProps, index) => ({
+        plotId: {
+          __custom: true,
+          title: plotProps.title ?? '',
+        } satisfies CustomPlotIdentifier,
+        isHidden: index >= DEFAULT_VISIBLE_PLOTS,
+        plotType: convertPlotlyPlotTypeToFacetNavPlotType(plotProps.type),
+      })),
+      ...(facetNavPanelPropsArray ?? []).map((facetPlotProps, index) => ({
+        plotId: facetPlotProps.facetToPlot,
+        isHidden: index >= DEFAULT_VISIBLE_PLOTS,
+        plotType: DEFAULT_PLOT_TYPE,
+      })),
+    ],
+    [],
+  )
+
+  useEffect(() => {
+    // const currentPlotIds = getCurrentPlotIds()
+    const combinedNewPlots = getCombinedNewPlots(
+      customPlots,
+      facetNavPanelPropsArray,
+    )
+
+    // Filter to only include new plots
+    const newPlots = combinedNewPlots.filter(plot => {
+      return !isPlotInState(plot.plotId)
+    })
+
+    // Update the state with new plots
+    setPlotUiStateArray(prevPlots => {
+      // Remove plots that are no longer in props from prevPlots
+      const isPlotStillPresent = (prevPlot: UiPlotState): boolean => {
+        if ('__custom' in prevPlot.plotId && prevPlot.plotId.__custom) {
+          const uniqueCustomPlot = prevPlot.plotId
+          return customPlots.some(
+            customPlot => customPlot.title === uniqueCustomPlot.title,
+          )
+        } else {
+          return facetNavPanelPropsArray.some(facetPlot =>
+            facetObjectMatchesDefinition(
+              facetPlot.facetToPlot,
+              prevPlot.plotId as UniqueFacetIdentifier,
+            ),
+          )
+        }
+      } // fn returns true iff the plot id is in customPlots or facetNavPanelPropsArray
+
+      const updatedProps = prevPlots.filter(isPlotStillPresent)
+
+      // Append new plots
+      const combinedPlots = [...updatedProps, ...newPlots]
+
+      // Remove duplicates
+      const uniquePlots = combinedPlots.filter(
```

Comment:
We should be able to remove this, let's use the debugger to figure out why we need it